### ISSUE TITLE
Example code doesn't compile for some reason

### ISSUE THREAD

**Issue by adsick** [rusqlite/rusqlite#884]
when I try to build it I get an error which says that something is wrong with [] parameters in the execute and query_map methods.

```
use rusqlite::{params, Connection, Result};

#[derive(Debug)]
struct Person {
    id: i32,
    name: String,
    data: Option<Vec<u8>>,
}


fn dbtest() -> Result<()> {
    let conn = Connection::open_in_memory()?;

    conn.execute(
        "CREATE TABLE person (
                  id              INTEGER PRIMARY KEY,
                  name            TEXT NOT NULL,
                  data            BLOB
                  )",
        [],
    )?;
    let me = Person {
        id: 0,
        name: "Steven".to_string(),
        data: None,
    };
    conn.execute(
        "INSERT INTO person (name, data) VALUES (?1, ?2)",
        params![me.name, me.data],
    )?;

    let mut stmt = conn.prepare("SELECT id, name, data FROM person")?;
    let person_iter = stmt.query_map([], |row| {
        Ok(Person {
            id: row.get(0)?,
            name: row.get(1)?,
            data: row.get(2)?,
        })
    })?;

    for person in person_iter {
        println!("Found person {:?}", person.unwrap());
    }
    Ok(())
}
```

error:

```
error[E0277]: `[_; 0]` is not an iterator
   --> src\app.rs:119:9
    |
119 |         [],
    |         ^^ borrow the array with `&` or call `.iter()` on it to iterate over it      
    |
    = help: the trait `Iterator` is not implemented for `[_; 0]`
    = note: arrays are not iterators, but slices like the following are: `&[1, 2, 3]`      
    = note: required because of the requirements on the impl of `IntoIterator` for `[_; 0]`

error[E0277]: `[_; 0]` is not an iterator
   --> src\app.rs:132:38
    |
132 |     let person_iter = stmt.query_map([], |row| {
    |                                      ^^ borrow the array with `&` or call `.iter()` on it to iterate over it
    |
    = help: the trait `Iterator` is not implemented for `[_; 0]`
    = note: arrays are not iterators, but slices like the following are: `&[1, 2, 3]`
    = note: required because of the requirements on the impl of `IntoIterator` for `[_; 0]`

error: aborting due to 2 previous errors; 1 warning emitted

For more information about this error, try `rustc --explain E0277`.
error: could not compile `egui_template`
```

pls. help me, it's quite disappointing.

**Comment by gwenn** [comment]
See https://github.com/rusqlite/rusqlite/pull/883 for an explanation.
And https://docs.rs/rusqlite/0.24.2/rusqlite/ or https://github.com/rusqlite/rusqlite/tree/v0.24.2 for examples matching the last released version...

**Comment by thomcc** [comment]
Also https://crates.io/crates/rusqlite shows the currently published version of the readme.

**Comment by thomcc** [comment]
(We should cut a release soon to avoid these, I should have some more time soon since some employment details of mine changed)

**Comment by adsick** [comment]
Ok, thx, hope I'll de able to make it work for me, I'm new to Rust.

**Comment by netroby** [comment]
For the master branch ,if you want the example code works, you should change your Cargo.toml like this
```
[dependencies]

[dependencies.rusqlite]
git="https://github.com/rusqlite/rusqlite"
branch="master"
features = ["bundled"]
```

**Comment by gwenn** [comment]
Should be fixed by version 0.25.0
